### PR TITLE
feat: add MPTCP support

### LINKED ISSUE #595: [Feature Request]  MPTCP 支持
### Greetings

_No response_

### Feature Request

各位大佬好 .
希望能增加 MPTCP 功能 , 已进内核 , 希望给Shadowsocks 和 V2RAY 之类的协议出站单独开启MPTCP .

### Use Cases

用于 LTE/5G/Starlink/VDSL 等线路聚合  . 目前ClashMeta , Xray-Core , Sing-box 已支持开启这个功能 .
用户仅需自行配置路由即可 .

### Potential Benefits

MPTCP是目前最佳的多线负载均衡以及网络备份方式 .

### PR DESCRIPTION
**Pull request by AkinoKaede**
### Background

Multipath TCP or MPTCP is an extension to the standard TCP and is described in RFC 8684. It allows a device to make use of multiple interfaces at once to send and receive TCP packets over a single MPTCP connection. MPTCP can aggregate the bandwidth of multiple interfaces or prefer the one with the lowest latency.

MPTCP is support in Go 1.21
 
### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOGS
- [ ] There is a user-facing docs PR against https://github.com/daeuniverse/dae

### Full Changelogs

- Add MPTCP support for nodes and subscription requests
- Update config description
- Update daeuniverse/outbound to latest

### Issue Reference

Closes #595

### Test Result
```bash
sudo ip mptcp monitor
```

image

### CODE REVIEW

File: config/desc.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ var GlobalDesc = Desc{
 	"sniffing_timeout":             "Timeout to waiting for first data sending for sniffing. It is always 0 if dial_mode is ip. Set it higher is useful in high latency LAN network.",
 	"tls_implementation":           "TLS implementation. \"tls\" is to use Go's crypto/tls. \"utls\" is to use uTLS, which can imitate browser's Client Hello.",
 	"utls_imitate":                 "The Client Hello ID for uTLS to imitate. This takes effect only if tls_implementation is utls. See more: https://github.com/daeuniverse/dae/blob/331fa23c16/component/outbound/transport/tls/utls.go#L17",
+	"mptcp":                        "Enable Multipath TCP.  It will take effects to all nodes.",
```
Comment (mzz2017) [inline config/desc.go:60]: 添加需要服务端支持的字样？

File: example.dae [hunk 2]
```diff
@@ -96,6 +96,9 @@ global {
     # The Client Hello ID for uTLS to imitate. This takes effect only if tls_implementation is utls.
     # See more: https://github.com/daeuniverse/dae/blob/331fa23c16/component/outbound/transport/tls/utls.go#L17
     utls_imitate: chrome_auto
+
+    # Multipath TCP (MPTCP) support. It will takes effect to all nodes.
```
Comment (mzz2017) [inline example.dae:100]: 同上

File: go.mod [hunk 3]
```diff
@@ -88,6 +87,8 @@ require (
 	google.golang.org/grpc v1.57.0 // indirect
 )
 
+replace github.com/daeuniverse/outbound v0.0.0-20240628165628-7c0c217530ea => github.com/AkinoKaede/outbound v0.0.0-20240807111200-47f3ba1371e2
```
Comment (mzz2017) [inline go.mod:90]: 这个是不是先合入

File: config/desc.go (hunk 1 above)
Comment (AkinoKaede) [reply config/desc.go]: I have modified it, but I'm not sure if the new description is accurate.

Comment (mzz2017) [reply config/desc.go]: 蛮好的，可以加入一些它的使用场景吗，例如：用作多接口或多ip之间的负载均衡

**Review by mzz2017 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**